src/Controller/SantaController.php

```diff
@@ -190,22 +215,35 @@ public function sendSampleMessage(Request $request, string $application): Respon
             });
         }
 
-        $message = trim($request->request->get('message', ''));
-        $notes = array_filter(array_map('trim', $request->request->all('notes')));
+        $session = $request->getSession();
+        $availableUsers = $session->get('available-users', []);
+        $selectedUsers = $session->get('selected-users', []);
+        $message = $session->get('message');
+
+        $form = $this->createForm(MessageType::class, null, [
+            'message' => $message,
+            'selected-users' => $selectedUsers,
+        ]);
+        $form->handleRequest($request);
 
-        if ($messageError = $this->validateMessage($message)) {
-            $errors['message'] = $messageError;
+        $notes = [];
+
+        $message = $form->getData()['message'] ?? '';
+        foreach ($form->getData() as $data => $note) {
+            if (str_contains($data, 'notes-') && $note) {
+                $notes[] = $note;
+            }
         }
 
-        if ($notesError = $this->validateNotes($notes)) {
-            $errors['notes'] = $notesError;
+        $formErrors = $form->getErrors(true) ?? null;
+
+        if (null !== $formErrors) {
+            foreach ($formErrors as $error) {
+                $errors[] = $error->getMessage();
+            }
```

Comment:
this iteration over error messages can be replaced by a call to `form->getErrors()`. See https://symfony.com/doc/current/components/form.html#accessing-form-errors for how to use it